### PR TITLE
add deposit agreement checkbox #374

### LINKED ISSUE #374: Avalon 6: Deposit checkbox
Previous work: 3c3dfa0

### PR DESCRIPTION
**Pull request by jefferya**
Fixes #374  ; refs https://github.com/ualbertalib/avalon/commit/3c3dfa03d363ccb87fca511ddc3642968e0d16d0

Adds to deposit form a checkbox confirming the users acknowledges the license agreement

The checkbox appears on the deposit item form just above the save buttons to acknowledge the user has: `I have read and do agree to the University of Alberta ERA Audio + Video Deposit and Distribution Agreement`. This functionality aligns with the Avalon 5 implementation here: https://github.com/ualbertalib/avalon/commit/3c3dfa03d363ccb87fca511ddc3642968e0d16d0

Changes proposed in this pull request:
* add checkbox and text to the form
* add javascript preventing the `save` and `save and continue` button operation unless confirmation of the license agreement
*

### CODE REVIEW

File: app/assets/javascripts/license_form.js [hunk 1]
```diff
@@ -15,4 +15,32 @@ $(document).ready(function() {
       $('#media_object_terms_of_use_custom').addClass('hidden');
     }
   });
+
```
Comment (cwant) [inline app/assets/javascripts/license_form.js]: Below the indentation looks inconsistent -- are tabs getting mixed with spaces?
Seems the file was using two spaces for indentation.

File: app/views/media_objects/_resource_description.html.erb [hunk 2]
```diff
@@ -145,6 +145,15 @@ Unless required by applicable law or agreed to in writing, software distributed
                                 dropdown_field: :note_type,
                                 dropdown_options: ModsDocument::NOTE_TYPES }} %>
 
+  <div class="form-group">
+    <input id="agreement_checkbox" type="checkbox" checked="false" name="deposit_agreement">
+      I have read and do agree to the
+      <a target="_blank" href="https://era.library.ualberta.ca/agreement/">
+	University of Alberta ERA Audio + Video Deposit and Distribution Agreement
```
Comment (cwant) [inline app/views/media_objects/_resource_description.html.erb]: This line could be indented a bit more to show that it's inside the `<a>`.

File: app/views/media_objects/_resource_description.html.erb [hunk 3]
```diff
@@ -145,6 +145,15 @@ Unless required by applicable law or agreed to in writing, software distributed
...
+      </a>
+    </input><br />
```
Comment (cwant) [inline app/views/media_objects/_resource_description.html.erb]: This line is red when viewed on github. I think `<input>` doesn't need a closing tag.

File: app/assets/javascripts/license_form.js (hunk 1 above)
Comment (jefferya) [reply app/assets/javascripts/license_form.js]: Changed.

File: app/views/media_objects/_resource_description.html.erb [hunk 4]
```diff
@@ -145,6 +145,16 @@ Unless required by applicable law or agreed to in writing, software distributed
...
 
+  <div class="form-group">
+    <label for="agreement_checkbox">
+      <input id="agreement_checkbox" type="checkbox" checked="false" name="deposit_agreement" />
+      I have read and do agree to the
+      <a target="_blank" href="https://era.library.ualberta.ca/agreement/">
+        University of Alberta ERA Audio + Video Deposit and Distribution Agreement.
+      </a>
+    </label>
+  </div>
```
Comment (jefferya) [inline app/views/media_objects/_resource_description.html.erb]: Changed indent and added label element for consistency.

**Pull request merged**